### DISCUSSION TITLE
How best to work with compressed optical discs (chds)?

### DISCUSSION THREAD

**Discussion by dirn** [emmercm/igir#1387] [Q&A]
My typical flow for an optical disc is to use MPF to dump the disc to bin and cue files. The filenames are often generic so I rename them using a redump dat file. Something like

    igir move --dat=SomeRedumpDatFile --input=/folder/with/bin/files --output=/folder/with/all/files

And rather than mucking around with the cue file, I keep all of the redump cue files in my input folder and let igir grab the appropriate one. So far, so good.

Before I ship the game off somewhere to play it, I compress it to chd. (I use chdman, but it would be awesome of igir could incorporate that step for me.)

Finally I copy (or link, doesn't matter) the games to a folder that I sync with whatever device I'm going to play on. This looks something like

    igir copy --dat=SomeRedumpDatFile --input=/folder/with/all/files --output=/folder/that/i/sync/{es}

Unfortunately this will copy the bin and cue files, leaving the chd files behind (the same applies to any m3u files I may have created). I've tried using dir2dat to generate a new dat file with only the files I actually want to sync, but the chd files are treated as the archives that they are and, instead of seeing a game with a single rom (the chd), I get a game with multiple roms (the bin(s) and cue).

Is there a way, either through move/copy/link or dir2dat (or some combination thereof) to have igir work with the chd files as singular roms rather than as archives? I'm not really sure what impact this would have on arcade romsets; I've only ever used igir with roms for consoles and handhelds.

**Comment by emmercm** [comment]
I'm not positive I understood the gap between expected behavior and actual behavior.

If I'm understanding right, it sounds like the input folder on your second command mixes duplicates in both compressed and uncompressed formats. Does `--input "/folder/with/all/files/**/*.chd"` solve your problem at all?
